utils/status_test.go

```diff
@@ -0,0 +1,315 @@
+	ssReplicasUnavailable := resourceCondition.GetReason()
+
+	statefulset.Status = appsv1.StatefulSetStatus{
+		Replicas:        3,
+		ReadyReplicas:   3,
+		UpdatedReplicas: 3,
+	}
+	r.GetClient().Status().Update(context.Background(), statefulset)
+
+	resourceCondition = r.areReplicasReady(runtimecomponent, newCondition)
+	ssReady := resourceCondition.GetReason()
+
+	// Test StatefulSet resource status
+	testSR := []Test{
+		{test: "StatefulSet not created", expected: "NotCreated", actual: ssNotCreated},
+		{test: "StatefulSet replicas unavailable", expected: "MinimumReplicasUnavailable", actual: ssReplicasUnavailable},
+		{test: "StatefulSet ready", expected: "MinimumReplicasAvailable", actual: ssReady},
+	}
+
+	verifyTests(testSR, t)
+}
+
+// Test isKnativeReady for Knative resource status check
+func TestKnativeReady(t *testing.T) {
+
+	// Setup fake client and reconciler base
+	// Set RuntimeComponent to use KnativeService
+	createKnativeService := true
+	spec = appstacksv1.RuntimeComponentSpec{Replicas: &st_replicas, CreateKnativeService: &createKnativeService}
+	r, runtimecomponent := setupFakeClientWithRC(spec)
+
+	// Report successful reconciliation to check for ResourcesReady condition
+	r.ManageSuccess(common.StatusConditionTypeReconciled, runtimecomponent)
+	conditionType := common.StatusConditionTypeResourcesReady
+	newCondition := runtimecomponent.GetStatus().NewCondition(conditionType)
+
+	// ResourcesReady condition should report ServiceNotCreated with no KnativeService created
+	resourceCondition := r.isKnativeReady(runtimecomponent, newCondition)
+	ksNotCreated := resourceCondition.GetReason()
+
+	// Create KnativeService and add the type to the fake client
+	// ResourcesReady condition should report ServiceStatusNotFound with empty KnativeService status
+	knative1 := &servingv1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+	}
+	r.scheme.AddKnownTypes(schema.GroupVersion{Group: "serving.knative.dev", Version: "v1"}, knative1)
+	r.CreateOrUpdate(knative1, runtimecomponent, func() error {
+		return nil
+	})
+	resourceCondition = r.isKnativeReady(runtimecomponent, newCondition)
+	ksStatusNotCreated := resourceCondition.GetReason()
+
+	// Unable to update status for KnativeService through fake client because it is not a known type
+	// Deleting and re-creating KnativeService to update its status conditions
+	// Update KnativeService with Not Ready condition
+	// ResourcesReady condition should report ServiceNotReady
+	r.DeleteResource(knative1)
+
+	knative2 := &servingv1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+	}
+	condition := apis.Condition{
+		Type:   apis.ConditionReady,
+		Status: v1.ConditionFalse,
+	}
+	conditions := apis.Conditions{condition}
+	knative2.Status.SetConditions(conditions)
+
+	r.scheme.AddKnownTypes(schema.GroupVersion{Group: "serving.knative.dev", Version: "v1"}, knative2)
```

Comment:
I'm not sure this needs to be done again (and a third time below). Not a huge issue, but I think it is just adding unnecessary code.